frontend/include/chpl/resolution/resolution-queries.h

```diff
@@ -113,7 +113,8 @@ ID lookupEnumElementByNumericValue(Context* context,
  */
 const TypedFnSignature*
 typedSignatureInitial(Context* context,
-                      const UntypedFnSignature* untypedSig);
+                      const UntypedFnSignature* untypedSig,
+                      const CallerDetails& caller={});
```

Comment:
We might want to use `const CallerDetails* caller=nullptr` if we think nested functions are more rare. Also might be nice to name the type or formal with something to indicate it's for nested functions.